.github/workflows/publish-package.yml

```diff
@@ -0,0 +1,88 @@
+name: Publish new package
+on:
+  workflow_dispatch:
+env:
+  PACKAGE_PREFIX: scottlogic-tech-carbon-estimator-
+concurrency:
+  group: 'publish'
+  cancel-in-progress: false
+jobs:
+  test:
+    uses: ./.github/workflows/pull-request.yml
+
+  package:
+    runs-on: ubuntu-latest
+    needs: test
+    outputs:
+      version: ${{ steps.output.outputs.version}}
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+      - name: Set up environment
+        uses: ./.github/actions/set-up-environment
+      - name: Create package
+        run: |
+          npm run prepare
+          cp README.md package.json ./dist/tech-carbon-estimator/
+          cd dist/tech-carbon-estimator
+          npm pkg delete scripts private devDependencies files
+          npm pkg set main=main.js
+          npm pack
+          VERSION=$(npm pkg get version --workspaces=false | tr -d \")
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          if-no-files-found: error
+          name: ${{env.PACKAGE_PREFIX}}${{env.VERSION}}
```

Comment:
I think this name could actually just be 'package' rather than including the full name/version, given the artefacts are specific to the workflow run. Only reduces the use of these env vars by a couple of places though and I don't know if it's nicer to have it named for its contents if you go looking at the workflow in the UI?

Reply:
I think either would be fine. I quite like seeing the name and version in the UI.

Reply:
Maybe a 'package_name' variable could help a bit with variable usage if that's easy to set up but I think it's fine like it is.

Reply:
Yeah, I think it's reasonably clear like this. If I set up another var it has to be added to this job, made an output of it, and included in the env section of each following job.